src/webui/api/synccontroller.cpp

```diff
@@ -369,13 +371,26 @@ namespace
 //  - "up_rate_limit: upload speed limit
 //  - "queueing": priority system usage flag
 //  - "refresh_interval": torrents table refresh interval
+//  - "free_space_on_disk": Free space on the default save path
 // GET param:
 //   - rid (int): last response id
 void SyncController::maindataAction()
 {
     auto lastResponse = sessionManager()->session()->getData(QLatin1String("syncMainDataLastResponse")).toMap();
     auto lastAcceptedResponse = sessionManager()->session()->getData(QLatin1String("syncMainDataLastAcceptedResponse")).toMap();
 
+    if (m_freeDiskSpaceThread == nullptr) {
+        m_freeDiskSpaceThread = new QThread(this);
+        FreeDiskSpaceChecker *freeDiskSpaceChecker = new FreeDiskSpaceChecker();
+        freeDiskSpaceChecker->moveToThread(m_freeDiskSpaceThread);
+        m_freeDiskSpaceThread->start();
+
+        m_freeDiskSpaceTimer = new QTimer(this);
```

Comment:
Is there any strong reason to have timer here (not in FreeDiskSpaceChecker)?

Reply:
Looks like you can do without a timer. Just keep the time of the last update and compare it with the current one. When the difference exceeds the specified timeout, call update().